app/components/post_textbox/post_textbox.test.js

```diff
@@ -336,4 +338,87 @@ describe('PostTextBox', () => {
             expect(wrapper.find(SendButton).prop('disabled')).toBe(true);
         });
     });
+
+    describe('Paste images', () => {
+        test('should show error dialog if error occured', () => {
+            jest.spyOn(Alert, 'alert').mockReturnValue(null);
+            const wrapper = shallowWithIntl(<PostTextbox {...baseProps}/>);
+            wrapper.find(PasteableTextInput).first().simulate('paste', {error: 'some error'}, []);
+            expect(Alert.alert).toHaveBeenCalled();
+        });
+
+        test('should show file max warning and not uploading', () => {
+            jest.spyOn(EventEmitter, 'emit').mockReturnValue(null);
+            const wrapper = shallowWithIntl(<PostTextbox {...baseProps}/>);
+            wrapper.find(PasteableTextInput).first().simulate('paste', null, [
+                {
+                    fileSize: 1000,
+                    fileName: 'fileName.png',
+                    type: 'images/png',
+                    url: 'path/to/image',
+                },
+                {
+                    fileSize: 1000,
+                    fileName: 'fileName.png',
+                    type: 'images/png',
+                    url: 'path/to/image',
+                },
+                {
+                    fileSize: 1000,
+                    fileName: 'fileName.png',
+                    type: 'images/png',
+                    url: 'path/to/image',
+                },
+                {
+                    fileSize: 1000,
+                    fileName: 'fileName.png',
+                    type: 'images/png',
+                    url: 'path/to/image',
+                },
+                {
+                    fileSize: 1000,
+                    fileName: 'fileName.png',
+                    type: 'images/png',
+                    url: 'path/to/image',
+                },
+                {
+                    fileSize: 1000,
+                    fileName: 'fileName.png',
+                    type: 'images/png',
+                    url: 'path/to/image',
+                },
+            ]);
+            expect(EventEmitter.emit).toHaveBeenCalledWith('fileMaxWarning');
+            expect(baseProps.actions.initUploadFiles).not.toHaveBeenCalled();
+        });
+
+        test('should show file size warning and not uploading', () => {
+            jest.spyOn(EventEmitter, 'emit').mockReturnValue(null);
+            const wrapper = shallowWithIntl(<PostTextbox {...baseProps}/>);
+            wrapper.find(PasteableTextInput).first().simulate('paste', null, [
+                {
+                    fileSize: 1500,
+                    fileName: 'fileName.png',
+                    type: 'images/png',
+                    url: 'path/to/image',
+                },
+            ]);
+            expect(EventEmitter.emit).toHaveBeenCalled();
+            expect(baseProps.actions.initUploadFiles).not.toHaveBeenCalled();
+        });
```

Comment:
nit: I suggest to test it like below so we can see the warning:
```javascript
test('should show file size warning and not uploading', () => {
    jest.spyOn(EventEmitter, 'emit').mockReturnValue(null);
    const wrapper = shallowWithIntl(
        <PostTextbox
            {...baseProps}
            maxFileSize={50 * 1024 * 1024}
        />
    );
    wrapper.find(PasteableTextInput).first().simulate('paste', null, [
        {
            fileSize: 51 * 1024 * 1024,
            fileName: 'fileName.png',
            type: 'images/png',
            url: 'path/to/image',
        },
    ]);

    expect(EventEmitter.emit).toHaveBeenCalledWith('fileSizeWarning', 'File above 50 MB cannot be uploaded: fileName.png');
    expect(baseProps.actions.initUploadFiles).not.toHaveBeenCalled();
});
```

Reply:
thanks for the suggestion, updated as here https://github.com/mattermost/mattermost-mobile/pull/3084/commits/c58e95a8af986af715575c979ffb80124a9cc1c6